### PR TITLE
Revised4012Q

### PR DESCRIPTION
**Pull request by PhilDeakin**
New code for 4012Q

### CODE REVIEW

**Comment by PhilDeakin** [comment]
I presume the lint thing is just saying it doesn't like how I've laid the code out.

**Comment by PhilDeakin** [comment]
@tab1tha Done. Thanks.

File: cin_validator/rules/cin2022_23/rule_4012Q.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        "01-01-2020",
+        "01-01-2020",
+        "01-03-2020",
+        "17-01-2020",
+        pd.NA,
+        "01-01-2020",
+        "15-01-2020",
+        "01-01-2020",
+    ]
+    #  Fails rows 0, 2, 6.
```
Comment (willLPD2I) [inline cin_validator/rules/cin2022_23/rule_4012Q.py:79]: Added for clarity

File: cin_validator/rules/cin2022_23/rule_4012Q.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+CINplanDates = CINTable.CINplanDates
+CINPlanStartDate = CINplanDates.CINPlanStartDate
+CINPlanEndDate = CINplanDates.CINPlanEndDate
+LAchildID = CINplanDates.LAchildID
+
+# define characteristics of rule
+@rule_definition(
+    code="4012Q",
+    rule_type=RuleType.QUERY,
```
Comment (willLPD2I) [inline cin_validator/rules/cin2022_23/rule_4012Q.py:17]: Set rule type so query

**Review by willLPD2I — APPROVED** [review]
Just added rule type to the rule_definiton decorator and added a comment to test_validate to explain which rows fail

**Pull request merged**